Revert "Remove device maps from TensorPipe for v1.7 release (#45353)"

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45762 Revert "Remove device maps from TensorPipe for v1.7 release (#45353)"**

This reverts commit 5211fb97ac4c246151f1286c78d63e0e317a8a4a.

Differential Revision: [D24088231](https://our.internmc.facebook.com/intern/diff/D24088231)

mrshenli committed:
Revert "Remove device maps from TensorPipe for v1.7 release (#45353)"

This reverts commit 5211fb97ac4c246151f1286c78d63e0e317a8a4a.

[ghstack-poisoned]

rohan-varma reviewed (approved):
LGTM as all this code was checked in before but removed for the 1.7 release.

mrshenli committed:
Update on "Revert "Remove device maps from TensorPipe for v1.7 release (#45353)""


This reverts commit 5211fb97ac4c246151f1286c78d63e0e317a8a4a.

Differential Revision: [D24088231](https://our.internmc.facebook.com/intern/diff/D24088231)

[ghstack-poisoned]
